test/test_c10d.py

```diff
@@ -247,23 +247,68 @@ def without(d, key):
             d.pop(key)
             return d
 
+        def withouts(d, keys):
+            d = d.copy()
+            for key in keys:
+                d.pop(key)
+            return d
+
         with Env(without(vars, 'WORLD_SIZE')):
             with self.assertRaisesRegex(ValueError, 'WORLD_SIZE expected'):
                 gen = c10d.rendezvous('env://')
                 next(gen)
+            c10d.init_process_group(backend='nccl', world_size=2)
+            self.assertEqual(c10d.get_rank(), 0)
+            self.assertEqual(c10d.get_world_size(), 2)
+            c10d.destroy_process_group()
+
```

Comment:
This fails when NCCL is not compiled. Can use gloo, or even better add the rank and world size kwargs to the rendezvous function and move this logic there. Then you don't have to init and destroy the process group, which you can't do without multiple processes when using the gloo backend.

I think this works for nccl because it only sets a key in the store, if it's rank is 0.

Reply:
I tried gloo, and it won't work for this case, since TCPstore will be written at PGGloo ctor, so the test will just hang. PGNccl ctr won't do this. So let's just use NCCL then. We don't want to test store here. Just would like to test if rank size are passed in correctly, that's it.

Reply:
Right, per my comment, it needs to be a multi process test case for gloo initialization to work. The test failures you see below are because NCCL isn't compiled in all configurations. Therefore, I suggested moving this logic into some function that takes the URL and kwargs and calls into rendezvous. Then you skip creating a process group entirely. This is the right approach here in my opinion as this test only tests getting the right rank and world size.

Reply:
I see you added a skip if not NCCL. This works but is a terrible hack. Can you create an issue to track fixing this the right way (I know there is some urgency here).